arkouda/matcher.py

```diff
@@ -113,10 +113,19 @@ def split(self, maxsplit: int = 0, return_segments: bool = False):
         if re.search(self.pattern, ""):
             raise ValueError("Cannot split or flatten with a pattern that matches the empty string")
         cmd = "segmentedSplit"
-        args = "{} {} {} {} {}".format(
-            self.objtype, self.parent_entry_name, maxsplit, return_segments, json.dumps([self.pattern])
+        repMsg = cast(
+            str,
+            generic_msg(
+                cmd=cmd,
+                args={
+                    "parent_name": self.parent_entry_name,
+                    "objtype": self.objtype,
+                    "max": maxsplit,
+                    "return_segs": return_segments,
+                    "pattern": [self.pattern],
+                },
```

Comment:
I was def just modeling after `flatten`. I think we can also just do `"pattern": self.pattern` here